.github/copilot-instructions.md

```diff
@@ -0,0 +1,267 @@
+# Copilot Instructions for Federation: 1999
+
+## Repository Overview
+
+This is a recreation of the Federation space fantasy game as it was in 1999. It's a **mixed-language project** combining Go (primary language), C, and C++ code. The game is a multi-user online text-based space game deployed to Fly.io.
+
+**Project Statistics:**
+- Primary languages: Go 1.25, C (gcc-13), C++ (g++-13)
+- Project size: ~7 main binaries (Go and C/C++)
+- Runtime: Linux (amd64), though non-Linux systems use Docker for building
+- Web frontend: Hugo 0.150.1 (extended) with Node.js/npm for dependencies
+
+## Critical Build Information
+
+### Build Order Dependencies
+
+**ALWAYS** follow this sequence when building from clean:
+
+1. **Generate** (required before any build or test):
+   ```bash
+   make generate
+   ```
+   This creates:
+   - `internal/server/parser/parser.tab.go` (using goyacc)
+   - `internal/text/messages.go` (using shell/awk/perl scripts)
+   
+   **Note**: Without `make generate`, builds and tests will fail with missing file errors.
+
+2. **Build** (platform-specific):
+   ```bash
+   make build
+   ```
+   - On Linux: Builds directly using system gcc/g++/go
+   - On non-Linux (macOS): Uses Docker with `Dockerfile.build` for cross-compilation
+   - Builds C/C++ libraries first (internal/ibgames, internal/fed), then binaries
+   - Output: Binaries in `bin/linux-amd64/`
+   
+   **Build time**: ~6 seconds on Linux for full build
+
+3. **Website** (optional, requires Hugo):
+   ```bash
+   make website
+   ```
+   - Requires Hugo 0.150.1 extended version installed
+   - First runs `npm ci` in web/, then `hugo build --minify`
+   - Hugo is NOT available in standard environments - check CI workflow for installation steps
+
+### Individual Build Targets
+
+**Go binaries** (built with `go build`):
+- `fedtpd` - Game engine (main server)
+- `httpd` - HTTP server
+- `login` - Authentication service
+- `perivale-go` - Go version of driver (incomplete)
```

Comment:
`perivale-go` is complete and mostly correct, the C version is just being preferred at the moment.